crates/meroctl/src/cli/auth.rs

```diff
@@ -32,19 +32,17 @@ pub async fn authenticate(api_url: &Url) -> Result<JwtToken> {
 
     // Set up callback server
     let (callback_port, callback_rx) = start_callback_server().await?;
-    println!("🌐 Started local callback server on port {}", callback_port);
 
     let auth_url = build_auth_url(api_url, callback_port)?;
-    println!("🔗 Opening browser to: {}", auth_url);
 
     if let Err(e) = webbrowser::open(&auth_url.to_string()) {
-        println!("⚠️  Failed to open browser automatically: {}", e);
-        println!("📋 Please manually open this URL in your browser:");
-        println!("   {}", auth_url);
+        bail!(
+            "Failed to open browser: {}. Please manually open this URL: {}",
+            e,
+            auth_url
+        );
```

Comment:
this will cause meroctl to exit right? but that's not what we want `{Info,Warn,Error}Line` works here